### DISCUSSION TITLE
Django rest framework: how to override `is_valid` on a serializer created with `many=True`?

### DISCUSSION THREAD

**Discussion by AnastasiusVivaldus** [encode/django-rest-framework#8353] [Question & Answer] [answered]
I have a puzzling question that I posted to [SO](https://stackoverflow.com/questions/70977197/django-rest-framework-how-to-override-is-valid-on-a-serializer-created-with) -- and would REALLY appreciate some help! Thanks!

**Comment by lovelydinosaur** [comment, marked as answer]
That's confusing isn't it?

There's a bit of a weird dance when you use `many=True`, which you can see in [the `__new__` method here](https://github.com/encode/django-rest-framework/blob/5bea22f32179276596b4c8152616d54a735a5341/rest_framework/serializers.py#L118).

The normal class creation is bypassed, and you end up with `MySerializer(many=True)` being equivalent to `ListSerializer(child=MySerializer())`.

Your custom `is_valid()` method will be called on any *items* in the list rather than on the list itself. I can't quite see how it'd lead to the exact behaviour describing but it might help you figure out how to move forward.

One thing that might help you continue to debug your issue would be to stop calling `get_serializer()` and instead instantiate the serializer directly in the `create` method. Then once you've done that, switch from `many=True` to using the explicit `ListSerializer(child...)` style instead. (Ie. gradually work towards less magic, and a more explicit bit of code behaviour until you've figured out exactly what's going on)

Hope that helps a little

**Reply by AnastasiusVivaldus** [reply]
Thanks very much! I did stumble on the [listserializer](https://www.django-rest-framework.org/api-guide/serializers/#listserializer) documentation after posting, and figured that had something to do with it.

I don't have the time/energy to dig into the DRF code to figure out exactly how this dance works but, in any case, after thinking harder about what exactly it is that I want, I realized it makes better sense when creating multiple instances of my model to go through each one within the `create` method, test the validity of each individual instance (with a separate serializer instance with `many=False` that uses my overriding `is_valid`) in order to just filter out the bad ones, rather than try to apply my overriding `is_valid` to the whole list; i.e. I'd prefer to create the good ones and ignore the bad ones, rather than have everything fail.
